src/plugins/index_pattern_editor/public/components/form_fields/timestamp_field.tsx

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+
+import { EuiFormRow, EuiComboBox, EuiFormHelpText, EuiComboBoxOptionOption } from '@elastic/eui';
+
+import {
+  UseField,
+  FieldConfig,
+  ValidationConfig,
+  getFieldValidityAndErrorMessage,
+} from '../../shared_imports';
+
+import { TimestampOption } from '../index_pattern_editor_flyout_content';
+import { schema } from '../form_schema';
+
+interface Props {
+  options: TimestampOption[];
+  isLoadingOptions: boolean;
+  isExistingIndexPattern: boolean;
+  isLoadingMatchedIndices: boolean;
+  hasMatchedIndices: boolean;
+}
+
+const requireTimestampOptionValidator = (options: Props['options']): ValidationConfig => ({
+  validator: async ({ value }) => {
+    const isValueRequired = !!options.length;
+    if (isValueRequired && !value) {
+      return {
+        message: i18n.translate(
+          'indexPatternEditor.requireTimestampOption.ValidationErrorMessage',
+          {
+            defaultMessage: 'Please select a timestamp field option.',
```

Comment:
Using "Please" is not something we have done in any of our form validation messages. Are you going to do a text copy review of this PR?

Reply:
Good call, I'll reach out to @gchaps

Reply:
Yes, "please" is not necessary here.  On another note, I see several forms of this message:

Field title:  Timestamp field
Hint text:  Select a primary time field for use with the global time filter
Validation: Select a timestamp field option
Validation:  Select a time stamp field

We should be consistent with the wording.  If "time field" is accurate, I'd go with that over "timestamp field".

Reply:
I think `timestamp` should be a property of an index pattern as I'm concerned that `time field` might be confused with the more general date type field. Also seems to match existing docs.